tools/cpp/unix_cc_toolchain_config.bzl

```diff
@@ -811,6 +811,24 @@ def _impl(ctx):
         ],
     )
 
+    is_linux = ctx.attr.target_libc != "macosx"
+    if is_linux:
+        versioned_library_flag_group = flag_group(
+            flags = ["-l:%{libraries_to_link.name}"],
+            expand_if_equal = variable_with_value(
+                name = "libraries_to_link.type",
+                value = "versioned_dynamic_library",
+            ),
+        )
+    else:
+        versioned_library_flag_group = flag_group(
+            flags = ["%{libraries_to_link.path}"],
```

Comment:
something like this change needs to happen in the osx_cc_toolchain_config too, and going forward we'll fix in the moved version in apple_support

Reply:
Done.